src/effects/effectsbackend.cpp

```diff
@@ -30,9 +31,12 @@ void EffectsBackend::registerEffect(const QString& id,
 
     const bool visible = m_pConfig->getValue<bool>(ConfigKey("[Visible Effects]", 
                                                    pManifest->id()), true);
```

Comment:
User preference code should be in the preference dialog classes, not scattered throughout the rest of the code. Reading the configuration values should be handled by DlgPrefEffects on initialization, which should make calls to EffectsManager::setEffectVisibility.